filebeat/crawler/prospector.go

```diff
@@ -110,6 +110,9 @@ func (p *Prospector) Run(wg *sync.WaitGroup) {
 			logp.Info("Run prospector")
 			p.prospectorer.Run()
 		}
+
+		// Defer next run for the defined scanFrequency
+		time.Sleep(p.ProspectorConfig.ScanFrequencyDuration)
```

Comment:
Nice solution, wasn't aware that this works :-) The only downside with this is that the first time the prospector is run is after the ScanFrequency. Could be solved by puttint a run before the for loop.

There is no change, the Sleep was part of `prospector_stdin` and `prospector_log` and I refactored it out.